src/domain/task-system/src/entities/task_state.rs

```diff
@@ -68,19 +68,18 @@ impl Projection for TaskState {
 
                 match event {
                     E::TaskCreated(_) => Err(ProjectionError::new(Some(s), event)),
-                    E::TaskRunning(TaskEventRunning {
-                        event_time,
-                        task_id: _,
-                    }) if s.status == TaskStatus::Queued => Ok(Self {
-                        status: TaskStatus::Running,
-                        ran_at: Some(event_time),
-                        ..s
-                    }),
-                    E::TaskCancelled(TaskEventCancelled {
-                        event_time,
-                        task_id: _,
-                    }) if s.status == TaskStatus::Queued
-                        || s.status == TaskStatus::Running && !s.cancellation_requested =>
+                    E::TaskRunning(TaskEventRunning { event_time, .. })
+                        if s.status == TaskStatus::Queued =>
+                    {
+                        Ok(Self {
+                            status: TaskStatus::Running,
+                            ran_at: Some(event_time),
+                            ..s
+                        })
+                    }
+                    E::TaskCancelled(TaskEventCancelled { event_time, .. })
```

Comment:
Particularly in the case of event projections I think the old style is much better to force you to consider projection when adding a field to an event.

Reply:
I actually started this change to avoid dummy fields in FlowsState. It's unlikely one will be changing events and not changing the state, this is a very local design decision.